pwnlib/elf/elf.py

```diff
@@ -769,7 +769,7 @@ def _patch_elf_and_read_maps(self):
             ...   with context.local(arch=arch):
             ...     sc = shellcraft.cat("/proc/self/maps")
             ...     sc += shellcraft.exit()
-            ...     sc = asm(sc)
+            ...     sc = asm(sc).rstrip(b'\x00\xbf')
```

Comment:
@Arusekk what is this rstrip thing here?

Reply:
Uh... it appends a nop instruction under thumb, but it is useless for `CAT_PROC_MAPS_EXIT`, since it takes two extra precious bytes; but I think we can add the nop to the preprocessed table as well for simplicity. Opening a new PR for these changes.